tests/unit/schema/test.firefox-schemas-import.js

```diff
@@ -1445,4 +1445,147 @@ describe('firefox schema import', () => {
+              },
+            ],
+          },
+        ],
+        expectedOutput: {
+          definitions: {
+            WebExtensionManifest: {
+              properties: {
+                page_action: {
+                  type: 'object',
+                  max_manifest_version: 2,
+                },
+              },
+            },
+          },
+        },
+      },
+      {
+        testName: 'propagates min_manifest_version',
+        schemaId: 'action',
+        schemaInput: [
+          {
+            namespace: 'manifest',
+            min_manifest_version: 3,
+            types: [
+              {
+                $extend: 'WebExtensionManifest',
+                properties: {
+                  action: {
+                    type: 'object',
+                  },
+                },
+              },
+            ],
+          },
+        ],
+        expectedOutput: {
+          definitions: {
+            WebExtensionManifest: {
+              properties: {
+                action: {
+                  type: 'object',
+                  min_manifest_version: 3,
+                },
+              },
+            },
+          },
+        },
+      },
+      {
+        testName: 'propagates min_manifest_version into a $ref property',
+        schemaId: 'fakeApi',
+        schemaInput: [
+          {
+            namespace: 'manifest',
+            min_manifest_version: 3,
+            types: [
+              {
+                $extend: 'WebExtensionManifest',
+                properties: {
+                  manifestPropWithRef: {
+                    $ref: 'SomeReferencedType',
+                  },
+                },
+              },
+            ],
+          },
+        ],
+        expectedOutput: {
+          definitions: {
+            WebExtensionManifest: {
+              properties: {
+                manifestPropWithRef: {
+                  allOf: [
+                    {
+                      $ref: 'manifest#/types/SomeReferencedType',
+                    },
+                    {
+                      min_manifest_version: 3,
+                    },
+                  ],
+                },
+              },
+            },
+          },
+        },
+      },
+      {
+        testName: 'does only propagates for api extended manifest definitions',
+        schemaId: 'fakeApi',
+        schemaInput: [
+          {
+            namespace: 'fakeApi',
+            min_manifest_version: 3,
+            types: [
+              {
+                $extend: 'SomeOtherType',
+                properties: {
+                  propName: { type: 'object' },
+                },
+              },
+            ],
+          },
+        ],
+        expectedOutput: {
+          definitions: {
+            SomeOtherType: {
+              properties: {
+                propName: { type: 'object' },
+              },
+            },
+          },
+        },
+      },
+    ];
+
+    for (const testCase of testCases) {
```

Comment:
When doing such things, I kinda like to have an extra assertion to make sure that all test cases will be taken into account:

```js
expect.assertions(testCases.length);
```

Otherwise, if you add `return` at the end of the loop code for example, tests would still pass.

Reply:
I keep forgetting about `it.each`, I actually really prefer to use that for this kind of test cases, I just need to remind myself of its existence :-P

Thanks for pointing it out ;-)